### DISCUSSION TITLE
eGPU works!! (No hotplug)

### DISCUSSION THREAD

**Discussion by sk0n3** [hackintosh-club/intel-nuc10#73] [Show and tell]
I got the egpu working consistently by combining
[NUC 8IxBEx's Thunderbolt SSDT](https://github.com/zearp/Nucintosh): (since thunderbolt drivers are the same)
SSDT-TYPC-NUC8-PC.aml

In SSDT-NUC10.aml file, I took everything after Scope (PSXS) all the way until Scope (LPCB) and then replaced it with the content of SSDT-TYPC-NUC8-PC.aml (After Scope (\_SB.PCI0.RP05.PXSX)). I don't know enough about this stuff so I didn't bother cleaning up the file after combining but maybe someone can do it?

**Comment by sk0n3** [comment]
<img width="392" alt="Screenshot 2022-07-24 at 6 37 15" src="https://user-images.githubusercontent.com/108772987/180649654-c316826f-16c6-4c40-90dd-4097e56fd427.png">
<img width="1072" alt="Screenshot 2022-07-24 at 6 37 30" src="https://user-images.githubusercontent.com/108772987/180649662-2de50bf4-082a-4b75-a0c0-3df53440e575.png">
<img width="1072" alt="Screenshot 2022-07-24 at 6 37 38" src="https://user-images.githubusercontent.com/108772987/180649665-ff60f4d1-0216-4cf7-8f14-4fd9f993ef3d.png">

**Reply by mrasong** [reply]
Did you try to do work like below?
1. Add `SSDT-TYPC-NUC8-BC.aml` to `/EFI/OC/ACPI` fold.
2. Add `SSDT-TYPC-NUC8-BC.aml` to NUC10's EFI config.list


![image](https://user-images.githubusercontent.com/1849527/180732513-cac6788a-24b5-4234-96f3-1185e741d2c6.png)

**Reply by mrasong** [reply]
DO NOT change `SSDT-NUC10.aml` and add extra `SSDT-TYPC-NUC8-BC.aml`

**Comment by sk0n3** [comment]
<img width="1920" alt="Screenshot 2022-07-25 at 1 35 20" src="https://user-images.githubusercontent.com/108772987/180734763-ed417bf2-f90e-43db-a04f-dc6d7e7b390a.png">
No not like that. I used Maciasl to replace thunderbolt ssdt

**Reply by sk0n3** [reply]
Does this make sense?

**Reply by mrasong** [reply]
Did this change affect other features of typec? eg. Typec to HDMI / DP? Typec hub?

**Reply by mrasong** [reply]
I can't test this cause I did not have an eGPU device.

**Reply by sk0n3** [reply]
Nothing that I notice as of right now, secondary type c display works, and my thunderbolt storage is also recognized too.

**Reply by sk0n3** [reply]
I have an owc miniStack STX thunderbolt dock and every port works just fine

**Reply by sk0n3** [reply]
I removed AGPM_NO.kext and substituted an appropriate power management kext using AGPMInjector

**Comment by lw8610** [comment]
Would it be possible for you to provide me the SSDT-NUC10.aml you created to get an eGPU work with NUC10? I would like to try it out.
